test/kafka-ingest-open-loop/mzcompose.py

```diff
@@ -0,0 +1,187 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import time
+
+import pg8000
+
+from materialize.mzcompose import Composition, WorkflowArgumentParser, say
+from materialize.mzcompose.services import (
+    Kafka,
+    Kgen,
+    Materialized,
+    Testdrive,
+    Zookeeper,
+)
+
+mz_persist = Materialized(
+    name="materialized_persist",
+    hostname="materialized",
+    workers=4,
+    timestamp_frequency="1s",
+    options="--persistent-user-tables --persistent-kafka-upsert-source --disable-persistent-system-tables-test",
+)
+
+mz_default = Materialized(workers=4, timestamp_frequency="1s")
+
+prerequisites = ["zookeeper", "kafka"]
+
+SERVICES = [
+    Zookeeper(),
+    Kafka(),
+    mz_default,
+    mz_persist,
+    # This instance of Testdrive is used to initialize the benchmark, and needs to have a
+    # known seed.
+    Testdrive(no_reset=True, seed=1),
+    Kgen(),
+]
```

Comment:
Can you refactor this to install `Materialized()` by default, and then use `c.override` to override in the desired definition based on the `--enable-persistence` flag? Here's an example: https://github.com/MaterializeInc/materialize/blob/209cd460d84ca2e8360adffa0d8ed857dfc70d2e/test/kafka-matrix/mzcompose.py#L40-L45

Then you can hardcode "materialized" everywhere instead of threading the service name around.

Reply:
nice! i like this a lot!